javaagent/sdk-migration-config.yaml

```diff
@@ -0,0 +1,291 @@
+            # Configure endpoint.
+            endpoint: ${OTEL_EXPORTER_OTLP_TRACES_ENDPOINT:-http://localhost:4318/v1/traces}
+            # Configure certificate.
+            certificate: ${OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE}
+            # Configure mTLS private client key.
+            client_key: ${OTEL_EXPORTER_OTLP_TRACES_CLIENT_KEY}
+            # Configure mTLS client certificate.
+            client_certificate: ${OTEL_EXPORTER_OTLP_TRACES_CLIENT_CERTIFICATE}
+            # Configure compression.
+            compression: ${OTEL_EXPORTER_OTLP_TRACES_COMPRESSION:-gzip}
+            # Configure max time (in milliseconds) to wait for each export.
+            timeout: ${OTEL_EXPORTER_OTLP_TRACES_TIMEOUT:-10000}
+            # Configure headers. Entries have higher priority than entries from .headers_list.
+            headers: []
+            # Configure headers. Entries have lower priority than entries from .headers.
+            # The value is a list of comma separated key-value pairs matching the format of OTEL_EXPORTER_OTLP_HEADERS. See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#configuration-options for details.
+            headers_list: ${OTEL_EXPORTER_OTLP_TRACES_HEADERS}
+  # Configure span limits. See also attribute_limits.
+  limits:
+    # Configure max attribute value size. Overrides .attribute_limits.attribute_value_length_limit.
+    attribute_value_length_limit: ${OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT}
+    # Configure max attribute count. Overrides .attribute_limits.attribute_count_limit.
+    attribute_count_limit: ${OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT:-128}
+    # Configure max span event count.
+    event_count_limit: ${OTEL_SPAN_EVENT_COUNT_LIMIT:-128}
+    # Configure max span link count.
+    link_count_limit: ${OTEL_SPAN_LINK_COUNT_LIMIT:-128}
+    # Configure max attributes per span event.
+    event_attribute_count_limit: ${OTEL_EVENT_ATTRIBUTE_COUNT_LIMIT:-128}
+    # Configure max attributes per span link.
+    link_attribute_count_limit: ${OTEL_LINK_ATTRIBUTE_COUNT_LIMIT:-128}
+  # Configure the sampler.
+  sampler:
+    # Configure sampler to be parent_based.
+    parent_based:
+      # Configure root sampler.
+      root:
+        # Configure sampler to be always_on.
+        always_on: {}
+      # Configure remote_parent_sampled sampler.
+      remote_parent_sampled:
+        # Configure sampler to be always_on.
+        always_on: {}
+      # Configure remote_parent_not_sampled sampler.
+      remote_parent_not_sampled:
+        # Configure sampler to be always_off.
+        always_off: {}
+      # Configure local_parent_sampled sampler.
+      local_parent_sampled:
+        # Configure sampler to be always_on.
+        always_on: {}
+      # Configure local_parent_not_sampled sampler.
+      local_parent_not_sampled:
+        # Configure sampler to be always_off.
+        always_off: {}
```

Comment:
this feels a little overwhelming for newbies, is it needed?

Reply:
It does and its also not strictly necessary since without explicitly setting these properties, the parent sampler is instantiated with these default values. This is a policy question (for the opentelemetry-configuration) project:

We have examples which are supposed to represent the default config, complete with env var substitution references to the env var config schema.

If a type (i.e. the parent sampler here) is part of the default config, should we enumerate all of its properties with their default values? Or should we limit the properties to just the required set? Or should we limit the properties to the required set, plus some (subjective) set of properties we deem will be common to configure?

Reply:
This comment prompted me to finally get around to fixing the empty objects (i.e. `always_off: {}` vs `always_off:`) upstream: https://github.com/open-telemetry/opentelemetry-configuration/pull/134